### PR TITLE
Duplicate time, Invalid time, enable button checks applied

### PR DESCRIPTION
**Pull request by SyedUsamaMeraj**

### CODE REVIEW

File: src/components/timeCard.vue [hunk 1]
```diff
@@ -1,76 +1,103 @@
...
       <div class="time-row">
         <div class="time-checkbox">
-          <select v-model="startTime" class="m-md-2 select" id="dropdown-1">
+          <select
+            v-model="startTime"
+            class="m-md-2 select"
+            id="startTime"
+            required
+          >
             <option selected disabled>Choose a time</option>
```
Comment (hamzamubeen-sl2) [inline src/components/timeCard.vue]: use this with value attribute set to empty string

File: src/components/timeCard.vue [hunk 2]
```diff
@@ -81,22 +108,52 @@
...
       endTime: "",
       times: [],
+      msg: "",
+      invalidTime: "",
     };
   },
   methods: {
     submitTime() {
-      this.times.push(this.startTime + " - " + this.endTime);
+      if (this.times.includes(this.startTime + " - " + this.endTime)) {
```
Comment (hamzamubeen-sl2) [inline src/components/timeCard.vue]: string literals

**Review by hamzamubeen-sl2 — CHANGES_REQUESTED** [review]

File: src/components/timeCard.vue [hunk 3]
```diff
@@ -86,18 +48,18 @@
       </div>
       <div class="mt-2 time-list">
         <div class="alert-msg">
-          <b-alert show dismissible v-if="msg">
-            {{ msg }}
+          <b-alert variant="danger" show dismissible v-if="sameTime">
```
Comment (hamzamubeen-sl2) [inline src/components/timeCard.vue]: larka 2 b-alerts ku use kar raha if ka bajai variant ka attribute toggle karwao

File: src/components/timeCard.vue [hunk 4]
```diff
@@ -5,72 +5,34 @@
...
-            <option>18:00</option>
-            <option>19:00</option>
-            <option>20:00</option>
-            <option>21:00</option>
-            <option>22:00</option>
-            <option>23:00</option>
+            <option
+              v-for="time in timeList"
+              :key="time.id"
+              :value="{ value: time.value, label: time.label }"
```
Comment (hamzamubeen-sl2) [inline src/components/timeCard.vue]: larka tumhain yahan ab label ki i zarrorat ni simple value ki ha. ku label tum baad main abb use to nahi kar raha

File: src/components/timeCard.vue [hunk 5]
```diff
@@ -86,18 +48,18 @@
...
         <div class="alert-msg">
           <b-alert variant="warning" show dismissible v-if="invalidTime">
-            {{ invalidTime }}
+            {{ errorMessage }}
           </b-alert>
         </div>
         <ul>
           <li v-for="(time, index) in times" :key="time.id">
-            <h5>{{ time }}</h5>
+            <h5>{{ time.startTime + " - " + time.endTime }}</h5>
```
Comment (hamzamubeen-rebel) [inline src/components/timeCard.vue:62]: or yahan pa time AM or PM main show karwao.

**Review by hamzamubeen-sl2 — CHANGES_REQUESTED** [review]

File: src/components/timeCard.vue (hunk 4 above)
Comment (SyedUsamaMeraj) [reply src/components/timeCard.vue]: is label ko neechay disabled ke property may use kea hai
kyu kay wo value ko consider nae kar raha hai.
or agar yahan say label get nae karta to neechay disabled function may label ke value get nae hoti

File: src/components/timeCard.vue [hunk 6]
```diff
@@ -1,75 +1,67 @@
...
+          <select
+            v-model="addTime.startTime"
+            class="m-md-2 select"
+            id="startTime"
+            required
+          >
+            <option
+              v-for="time in timeList"
+              :key="time.id"
+              :value="{ value: time.value  }"
```
Comment (hamzamubeen-sl2) [inline src/components/timeCard.vue]: cant we just do it like :value="time.value"

File: src/components/daysList.vue [hunk 7]
```diff
@@ -26,18 +25,30 @@ export default {
...
   },
+  methods: {
+    sortDay: function (day) {
+      return day.slice().sort(function (a, b) {
+        return a.value - b.value;
+      });
+    },
+    addDays() {
+      this.selectedDays.push(
+        { value: this.daylist.value },
```
Comment (hamzamubeen-sl2) [inline src/components/daysList.vue]: you are pushing two seperate objects into the array

File: src/components/timeCard.vue [hunk 8]
```diff
@@ -81,22 +68,135 @@
...
+        { value: 19, label: "07:00 PM" },
+        { value: 20, label: "08:00 PM" },
+        { value: 21, label: "09:00 PM" },
+        { value: 22, label: "10:00 PM" },
+        { value: 23, label: "11:00 PM" },
+      ],
     };
   },
   methods: {
+    sortTime: function (time) {
```
Comment (hamzamubeen-sl2) [inline src/components/timeCard.vue]: try and use ternary operators here instead of if else

File: src/components/timeCard.vue [hunk 9]
```diff
@@ -81,22 +68,135 @@
...
+        } else {
+          return 0;
+        }
+      });
+    },
+    convertTime(time) {
+      let addedTime = time;
+      let prefix = "AM";
+      let hours = addedTime;
+      if (hours >= 12) {
```
Comment (hamzamubeen-sl2) [inline src/components/timeCard.vue:106]: R&D check if it is better to use two if's or one if else

File: src/components/timeCard.vue [hunk 10]
```diff
@@ -81,22 +68,135 @@
...
+        this.invalidTime = false;
+        this.sameTime = false;
+      }
     },
     deleteTime(index) {
       this.times.splice(index, 1);
     },
   },
+  computed: {
+    selectDisabled() {
```
Comment (hamzamubeen-sl2) [inline src/components/timeCard.vue:152]: directly return the results

**Review by hamzamubeen-sl2 — CHANGES_REQUESTED** [review]

**Review by hamzamubeen-sl2 — APPROVED** [review]

**Pull request merged**